### PR TITLE
Updated regex that filters IP addresses in CES

### PR DESCRIPTION
**Pull request by shubh1410**
updated regex to handle IPv6/IPv4 with prefix/subnet

Signed-off-by: Shubham Darokar(shubham.dipak.darokar1@ibm.com)

### CODE REVIEW

File: roles/nfs/common/vars/main.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@
...
 
-scale_ipv4_regex: ^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$
+scale_ipv6_regex: ^s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|(([0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(:(((:[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:)))(%.+)?s*(\/([0-9]|[1-9][0-9]|1[0-1][0-9]|12[0-8]))?$
```
Comment (acch) [inline roles/nfs/common/vars/main.yml:8]: I really hope that I don't ever have to debug this code ;-)

Maybe it's worth adding a few samples to help your future self understand what this code does?

Comment (shubh1410) [reply roles/nfs/common/vars/main.yml:8]: Sorry, I don't understand what you mean. It is not a code. It is just a variable and the string value is regular expression that filters IPv6 addresses.

**Comment by acch** [comment]
Hi @shubh1410 - thanks for the explanation. You are right, I'm worried about the complexity of the regex.

Ansible has a [Jinja2 filter](https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters_ipaddr.html) `ipaddr()` that allows to test various data to check if they are valid IP addresses, and also allows to query and manipulate the data. It supports IPv4 and IPv6 and looks pretty comprehensive:

https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters_ipaddr.html

Can we use the `ipaddr()` filter instead of maintaining our own regex?

**Comment by shubh1410** [comment]
@acch -- ipaddr() requires python library 'netaddr' which needs to be installed. Hence we did not use the filter to reduce the dependancy.

**Comment by Christop1964** [comment]
I've tested the new regex with IPv4 and IPv6 and it works as expected. 
So for me it looks fine !

**Review by rajan-mis — APPROVED** [review]
We were thinking to use ipaddr() but then we will always have to install this package. @acch Thanks for suggestion, we can discuss this in our discussion.  for now , i am thinking to merge to unblock verification.

**Pull request merged**